cmd/mdatagen/internal/event.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/cmd/mdatagen/internal"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type (
+	EventName string
+)
+
+func (ln EventName) Render() (string, error) {
+	return FormatIdentifier(string(ln), true)
+}
+
+func (ln EventName) RenderUnexported() (string, error) {
+	return FormatIdentifier(string(ln), false)
+}
+
+type Event struct {
+	// Enabled defines whether the event is enabled by default.
+	Enabled bool `mapstructure:"enabled"`
+
+	// Warnings that will be shown to user under specified conditions.
+	Warnings Warnings `mapstructure:"warnings"`
+
+	// Body stores metadata for event body
+	Body ValueType `mapstructure:"body"`
```

Comment:
Why `ValueType`? The logs body can be anything. However, do structured events need it at all?

Reply:
Remove this field, as it may not be necessary. I can add it later if some use cases arise.